### ISSUE TITLE
screenreader reading of inline errors

### ISSUE THREAD

**Issue by albertotirla** [microsoft/vscode#148634]
Issue Type: <b>Feature Request</b>

We already have audio queues for when the focus goes over a line that contains errors, perhaps to give us the same experience sighted people have, where they see the errors underlined in red I believe. Next step, the same way sighted people can mouse over the portion of code underlined in red to see a tooltip with the error message, we should also get the same information either by explicitly pressing a key, or through the mechanism that's already used for parameter hints

# how should it work?

1. Open a file in vscode, the language should be detectable and an extenssion that provides the language server should be installed
2. If there isn't any error, make one
3. With the arrow keys, go over that line. If everything is working right, you should hear the tone ment for errors

now, you could either wait a bit longer while on that line to hear the error text if using the parameter hint mechanism, or press a key command to do the same thing if you think the params hint way is too noisy and you want the users to have explicit controll over when the screenreader reads the error content.  
yes, I know I could use the problems view, but sometimes I have so many errors and warnings from other files as well, that the error I'm literarely focused on is lost in the noise.  
I could also use the mouseover command, but that only works as long as my cursor is focused on the part of the line where the underlining in red begins. Since I don't have any way to know where that is, I can't use that feature as it is. Furthermore, the mouseover command pops up a tooltip which moves the focus of the screenreader, requiring me to press the esc key to get back to the editor, so going through the problems view or issuing the next/previous problem commands would end up being more efficient.

VS Code version: Code 1.66.2 (dfd34e8260c270da74b5c2d86d61aee4b6d56977, 2022-04-11T07:46:01.075Z)
OS version: Windows_NT x64 10.0.19044
Restricted Mode: No

**Comment by albertotirla** [comment]
@isidorn this has been discussed in the vscode a11y matrix room. There, you told me to ping you when I make the issue, so perhaps now is the time to do so

**Comment by albertotirla** [comment]
the problem with f8 is that it could present me with what it thinks is the next error/problem, which may be far away from the line I was focused on. I could issue next problem followed by previous problem, but that's not guaranteed to bring me where I want, talking from experience. Before I forget, next problem and previous problem have a habit of searching even through other files, for things like warnings, real errors most likely being able to then hide in the noise, so the issue is still not solved.

**Comment by isidorn** [comment]
@albertotirla thanks for the feedback. What if we changed the F8 to always first go to the current line, even if the column is before your column. Would that help?

**Comment by albertotirla** [comment]
> @albertotirla thanks for the feedback. What if we changed the F8 to always first go to the current line, even if the column is before your column. Would that help?

yes, it would. Also, it would be nice if the cursor would move to where the error is on the line, selecting it possibly. If no other option can be found, then this is a good enough way of doing it

**Comment by isidorn** [comment]
@lszomoru my last suggestion sounds low hanging and might be a win. What do you think?

Also @jrieken for thoughts

**Comment by jrieken** [comment]
👋
